list.c

```diff
@@ -0,0 +1,13 @@
+#include "list.h"
+
+int hash(char* key) {
+    int hashIndex = 0;
+    while (key[hashIndex] != '\0') {
+        hashIndex++;
+    }
+    return hashIndex;
+}
+
+int getValueHash(char* key) {
```

Comment:
```suggestion
int getValueHash(const char* key) {
```
Внутри параметр меняться по смыслу не должен, и это даст возможность вызывать функцию от константных строк, например, литералов.